### ISSUE TITLE
Dismissible ModalBarrier needs an a11y label

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#13451]
On iOS VoiceOver currently reads out garbage when a11y focusing on the ModalBarrier because it doesn't have a proper label.

Android doesn't have the problem because on this platform the barrier is not a11y focusable (the back button is used to dismiss).

**Comment by goderbauer** [comment]
Adding a label is not as straight forward as it seems. ModalBarrier lives in the widget layer, which is not supposed to have internationalized strings. The proposed solution is to add a new String property `barrierLabel` to `ModalRoute`.
